api/fixtures/example.go

```diff
@@ -12,19 +12,19 @@ import (
 	crclient "sigs.k8s.io/controller-runtime/pkg/client"
 )
 
-type ExampleResources struct {
-	Namespace    *corev1.Namespace
-	PullSecret   *corev1.Secret
-	AWSResources *ExampleAWSResources
-	SSHKey       *corev1.Secret
-	Cluster      *hyperv1.HostedCluster
-	NodePool     *hyperv1.NodePool
+type Resources interface {
+	AsObjects() []crclient.Object
 }
 
-type ExampleAWSResources struct {
-	KubeCloudControllerAWSCreds  *corev1.Secret
-	NodePoolManagementAWSCreds   *corev1.Secret
-	ControlPlaneOperatorAWSCreds *corev1.Secret
+var _ Resources = &ExampleResources{}
```

Comment:
Why does `ExampleResources` need to implement the `Resources` interface? It only seems to be necesary for the `ExampleAWSResources`

Reply:
that's because even `ExampleResources` is also the type of `Resources` and also got `AsObjects` method to dump the content.

Reply:
I can see that, but whatever calls the `AsObjects` doesn't take a `Resources` interface so this seems like a premature optimization?